File: app/src/main/java/space/pal/sig/view/web/WebActivity.java
package space.pal.sig.view.web;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.webkit.WebChromeClient;
import android.webkit.WebView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;

import butterknife.BindView;
import space.pal.sig.R;
import space.pal.sig.view.SpaceBaseActivity;

/**
 * SpaceNow
 * Created by Catalin on 7/28/2020
 **/
public class WebActivity extends SpaceBaseActivity {

    public static final String SPACE_URL = "url";

    @BindView(R.id.web_view)
    WebView webView;
    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @SuppressLint("SetJavaScriptEnabled")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeAsUpIndicator(R.drawable.ic_baseline_close_24);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
        String url = getIntent().getStringExtra(SPACE_URL);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebChromeClient(new WebChromeClient());
        if (url != null) {
            webView.loadUrl(url);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return super.onSupportNavigateUp();
    }
}

File: app/src/main/java/space/pal/sig/view/extra/WebActivity.kt
package space.pal.sig.view.extra

import android.os.Bundle
import androidx.core.view.isVisible
import androidx.databinding.DataBindingUtil
import space.pal.sig.R
import space.pal.sig.databinding.ActivityWebBinding
import space.pal.sig.util.CustomWebViewClient
import space.pal.sig.view.BaseActivity

class WebActivity : BaseActivity() {

    companion object {
        const val URL = "url"
        const val TITLE = "title"
    }

    private lateinit var binding: ActivityWebBinding

    private val loadedRunnable = Runnable {
        binding.webView.isVisible = true
        binding.webLoading.isVisible = false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_web)
        setSupportActionBar(binding.webToolbar)
        supportActionBar?.let {
            it.setDisplayHomeAsUpEnabled(true)
            it.setHomeAsUpIndicator(R.drawable.ic_baseline_close_24)
        }
        val url = intent.getStringExtra(URL)
        val title = intent.getStringExtra(TITLE)
        this.title = title
        setupWebView(url)
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return super.onSupportNavigateUp()
    }

    private fun setupWebView(url: String?) {
        binding.webView.settings.javaScriptEnabled = true
        url?.let {
            binding.webView.webViewClient = CustomWebViewClient(loadedRunnable)
            binding.webView.loadUrl(it)
        }
    }
}